core/watcher/eventsource/namespace.go

```diff
@@ -97,6 +120,17 @@ func (w *NamespaceWatcher) loop() error {
 				return nil
 			}
 
+			// Check with the predicate to determine if we should send a
+			// notification.
+			ctx := w.tomb.Context(context.Background())
```

Comment:
Probably should avoid this context on every change.

Reply:
Normally I would agree, but this should be called relatively lightly. I'd be very surprised if this is even called once a day.

Reply:
Actually, we could just cache this one. Good spot @hpidcock

Reply:
I've done it anyway, it's free.